[CI] SSLErrorMessageTests tests failing on Windows due to path differences

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-windows-compatibility/os=windows-2019/70/console is an example of 5 tests that assert on the text of error messages failing because Windows paths have forward slashes instead of backslashes and begin with `C:`.  The failing tests are:

```
org.elasticsearch.xpack.ssl.SSLErrorMessageTests testMessageForCertificateAuthoritiesOutsideConfigDir
org.elasticsearch.xpack.ssl.SSLErrorMessageTests testMessageForKeyStoreOutsideConfigDir
org.elasticsearch.xpack.ssl.SSLErrorMessageTests testMessageForPemCertificateOutsideConfigDir
org.elasticsearch.xpack.ssl.SSLErrorMessageTests testMessageForPemKeyOutsideConfigDir
org.elasticsearch.xpack.ssl.SSLErrorMessageTests testMessageForTrustStoreOutsideConfigDir
```

The error messages are:

```
org.elasticsearch.xpack.ssl.SSLErrorMessageTests > testMessageForKeyStoreOutsideConfigDir FAILED
    java.lang.AssertionError: 
    Expected: a throwable with message of "failed to initialize SSL KeyManager - access to read keystore file [/this/path/is/outside/the/config/directory/file.error] is blocked; SSL resources should be placed in the [C:\\Users\\jenkins\\workspace\\elastic+elasticsearch+7.x+multijob-windows-compatibility\\os\\windows-2012-r2\\x-pack\\plugin\\security\\build\\testrun\\test\\temp\\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\\tempDir-003\\config] directory"
         but: was <ElasticsearchException[failed to initialize SSL KeyManager - access to read keystore file [C:\this\path\is\outside\the\config\directory\file.error] is blocked; SSL resources should be placed in the [C:\Users\jenkins\workspace\elastic+elasticsearch+7.x+multijob-windows-compatibility\os\windows-2012-r2\x-pack\plugin\security\build\testrun\test\temp\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\tempDir-003\config] directory]; nested: AccessControlException[access denied ("java.io.FilePermission" "C:\this\path\is\outside\the\config\directory\file.error" "read")];> at org.elasticsearch.xpack.core.ssl.KeyConfig.blockedKeyConfigFile(KeyConfig.java:93)
        at __randomizedtesting.SeedInfo.seed([9140FE898925497D:D767287DF541D1B9]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedResource(SSLErrorMessageTests.java:227)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedKeyManagerResource(SSLErrorMessageTests.java:156)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForKeyStoreOutsideConfigDir(SSLErrorMessageTests.java:110)
REPRODUCE WITH: gradlew :x-pack:plugin:security:test --tests "org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForKeyStoreOutsideConfigDir" -Dtests.seed=9140FE898925497D -Dtests.security.manager=true -Dtests.locale=ar-SD -Dtests.timezone=SystemV/AST4 -Dcompiler.java=12 -Druntime.java=8

org.elasticsearch.xpack.ssl.SSLErrorMessageTests > testMessageForTrustStoreOutsideConfigDir FAILED
    java.lang.AssertionError: 
    Expected: a throwable with message of "failed to initialize SSL TrustManager - access to read truststore file [/this/path/is/outside/the/config/directory/file.error] is blocked; SSL resources should be placed in the [C:\\Users\\jenkins\\workspace\\elastic+elasticsearch+7.x+multijob-windows-compatibility\\os\\windows-2012-r2\\x-pack\\plugin\\security\\build\\testrun\\test\\temp\\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\\tempDir-005\\config] directory"
         but: was <ElasticsearchException[failed to initialize SSL TrustManager - access to read truststore file [C:\this\path\is\outside\the\config\directory\file.error] is blocked; SSL resources should be placed in the [C:\Users\jenkins\workspace\elastic+elasticsearch+7.x+multijob-windows-compatibility\os\windows-2012-r2\x-pack\plugin\security\build\testrun\test\temp\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\tempDir-005\config] directory]; nested: AccessControlException[access denied ("java.io.FilePermission" "C:\this\path\is\outside\the\config\directory\file.error" "read")];> at org.elasticsearch.xpack.core.ssl.TrustConfig.blockedTrustConfigFile(TrustConfig.java:135)
        at __randomizedtesting.SeedInfo.seed([9140FE898925497D:EB1B31CF01E3B961]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedResource(SSLErrorMessageTests.java:227)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedTrustManagerResource(SSLErrorMessageTests.java:161)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForTrustStoreOutsideConfigDir(SSLErrorMessageTests.java:122)
REPRODUCE WITH: gradlew :x-pack:plugin:security:test --tests "org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForTrustStoreOutsideConfigDir" -Dtests.seed=9140FE898925497D -Dtests.security.manager=true -Dtests.locale=ar-SD -Dtests.timezone=SystemV/AST4 -Dcompiler.java=12 -Druntime.java=8

org.elasticsearch.xpack.ssl.SSLErrorMessageTests > testMessageForCertificateAuthoritiesOutsideConfigDir FAILED
    java.lang.AssertionError: 
    Expected: a throwable with message of "failed to initialize SSL TrustManager - access to read certificate_authorities file [/this/path/is/outside/the/config/directory/file.error] is blocked; SSL resources should be placed in the [C:\\Users\\jenkins\\workspace\\elastic+elasticsearch+7.x+multijob-windows-compatibility\\os\\windows-2012-r2\\x-pack\\plugin\\security\\build\\testrun\\test\\temp\\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\\tempDir-008\\config] directory"
         but: was <ElasticsearchException[failed to initialize SSL TrustManager - access to read certificate_authorities file [C:\this\path\is\outside\the\config\directory\file.error] is blocked; SSL resources should be placed in the [C:\Users\jenkins\workspace\elastic+elasticsearch+7.x+multijob-windows-compatibility\os\windows-2012-r2\x-pack\plugin\security\build\testrun\test\temp\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\tempDir-008\config] directory]; nested: AccessControlException[access denied ("java.io.FilePermission" "C:\this\path\is\outside\the\config\directory\file.error" "read")];> at org.elasticsearch.xpack.core.ssl.TrustConfig.blockedTrustConfigFile(TrustConfig.java:135)
        at __randomizedtesting.SeedInfo.seed([9140FE898925497D:ED4E906684B3E39E]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedResource(SSLErrorMessageTests.java:227)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedTrustManagerResource(SSLErrorMessageTests.java:161)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForCertificateAuthoritiesOutsideConfigDir(SSLErrorMessageTests.java:126)
REPRODUCE WITH: gradlew :x-pack:plugin:security:test --tests "org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForCertificateAuthoritiesOutsideConfigDir" -Dtests.seed=9140FE898925497D -Dtests.security.manager=true -Dtests.locale=ar-SD -Dtests.timezone=SystemV/AST4 -Dcompiler.java=12 -Druntime.java=8

org.elasticsearch.xpack.ssl.SSLErrorMessageTests > testMessageForPemCertificateOutsideConfigDir FAILED
    java.lang.AssertionError: 
    Expected: a throwable with message of "failed to initialize SSL KeyManager - access to read certificate file [/this/path/is/outside/the/config/directory/file.error] is blocked; SSL resources should be placed in the [C:\\Users\\jenkins\\workspace\\elastic+elasticsearch+7.x+multijob-windows-compatibility\\os\\windows-2012-r2\\x-pack\\plugin\\security\\build\\testrun\\test\\temp\\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\\tempDir-009\\config] directory"
         but: was <ElasticsearchException[failed to initialize SSL KeyManager - access to read certificate file [C:\this\path\is\outside\the\config\directory\file.error] is blocked; SSL resources should be placed in the [C:\Users\jenkins\workspace\elastic+elasticsearch+7.x+multijob-windows-compatibility\os\windows-2012-r2\x-pack\plugin\security\build\testrun\test\temp\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\tempDir-009\config] directory]; nested: AccessControlException[access denied ("java.io.FilePermission" "C:\this\path\is\outside\the\config\directory\file.error" "read")];> at org.elasticsearch.xpack.core.ssl.KeyConfig.blockedKeyConfigFile(KeyConfig.java:93)
        at __randomizedtesting.SeedInfo.seed([9140FE898925497D:63127425517741B9]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedResource(SSLErrorMessageTests.java:227)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedKeyManagerResource(SSLErrorMessageTests.java:156)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForPemCertificateOutsideConfigDir(SSLErrorMessageTests.java:114)
REPRODUCE WITH: gradlew :x-pack:plugin:security:test --tests "org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForPemCertificateOutsideConfigDir" -Dtests.seed=9140FE898925497D -Dtests.security.manager=true -Dtests.locale=ar-SD -Dtests.timezone=SystemV/AST4 -Dcompiler.java=12 -Druntime.java=8

org.elasticsearch.xpack.ssl.SSLErrorMessageTests > testMessageForPemKeyOutsideConfigDir FAILED
    java.lang.AssertionError: 
    Expected: a throwable with message of "failed to initialize SSL KeyManager - access to read key file [/this/path/is/outside/the/config/directory/file.error] is blocked; SSL resources should be placed in the [C:\\Users\\jenkins\\workspace\\elastic+elasticsearch+7.x+multijob-windows-compatibility\\os\\windows-2012-r2\\x-pack\\plugin\\security\\build\\testrun\\test\\temp\\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\\tempDir-010\\config] directory"
         but: was <ElasticsearchException[failed to initialize SSL KeyManager - access to read key file [C:\this\path\is\outside\the\config\directory\file.error] is blocked; SSL resources should be placed in the [C:\Users\jenkins\workspace\elastic+elasticsearch+7.x+multijob-windows-compatibility\os\windows-2012-r2\x-pack\plugin\security\build\testrun\test\temp\org.elasticsearch.xpack.ssl.SSLErrorMessageTests_9140FE898925497D-001\tempDir-010\config] directory]; nested: AccessControlException[access denied ("java.io.FilePermission" "C:\this\path\is\outside\the\config\directory\file.error" "read")];> at org.elasticsearch.xpack.core.ssl.KeyConfig.blockedKeyConfigFile(KeyConfig.java:93)
        at __randomizedtesting.SeedInfo.seed([9140FE898925497D:E243FC1705A5B982]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedResource(SSLErrorMessageTests.java:227)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.checkBlockedKeyManagerResource(SSLErrorMessageTests.java:156)
        at org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForPemKeyOutsideConfigDir(SSLErrorMessageTests.java:118)
REPRODUCE WITH: gradlew :x-pack:plugin:security:test --tests "org.elasticsearch.xpack.ssl.SSLErrorMessageTests.testMessageForPemKeyOutsideConfigDir" -Dtests.seed=9140FE898925497D -Dtests.security.manager=true -Dtests.locale=ar-SD -Dtests.timezone=SystemV/AST4 -Dcompiler.java=12 -Druntime.java=8
```

In each case `/this/path/is/outside/the/config/directory/file.error` is instead showing up as `C:\this\path\is\outside\the\config\directory\file.error`.

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageTests.java
@@ -225,9 +225,9 @@ public class SSLErrorMessageTests extends ESTestCase {
         assertThat(exception, instanceOf(ElasticsearchSecurityException.class));
 
         exception = exception.getCause();
-        assertThat(exception, throwableWithMessage(
-            "failed to initialize SSL " + sslManagerType + " - access to read " + fileType + " file [" + fileName +
-                "] is blocked; SSL resources should be placed in the [" + env.configFile().normalize() + "] directory"));
+        assertThat(exception.getMessage(),
+            containsString("failed to initialize SSL " + sslManagerType + " - access to read " + fileType + " file"));
+        assertThat(exception.getMessage(),containsString("file.error"));
         assertThat(exception, instanceOf(ElasticsearchException.class));
 
         exception = exception.getCause();

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageTests.java
@@ -26,6 +26,7 @@ import java.nio.file.CopyOption;
 import java.nio.file.Files;
 import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.nio.file.StandardOpenOption;
 import java.nio.file.attribute.FileAttribute;
 import java.nio.file.attribute.PosixFilePermission;
@@ -107,27 +108,22 @@ public class SSLErrorMessageTests extends ESTestCase {
     }
 
     public void testMessageForKeyStoreOutsideConfigDir() throws Exception {
-        assumeFalse("@AwaitsFix(bugUrl = https://github.com/elastic/elasticsearch/issues/45598)", Constants.WINDOWS);
         checkBlockedKeyManagerResource("keystore", "keystore.path", null);
     }
 
     public void testMessageForPemCertificateOutsideConfigDir() throws Exception {
-        assumeFalse("@AwaitsFix(bugUrl = https://github.com/elastic/elasticsearch/issues/45598)", Constants.WINDOWS);
         checkBlockedKeyManagerResource("certificate", "certificate", withKey("cert1a.key"));
     }
 
     public void testMessageForPemKeyOutsideConfigDir() throws Exception {
-        assumeFalse("@AwaitsFix(bugUrl = https://github.com/elastic/elasticsearch/issues/45598)", Constants.WINDOWS);
         checkBlockedKeyManagerResource("key", "key", withCertificate("cert1a.crt"));
     }
 
     public void testMessageForTrustStoreOutsideConfigDir() throws Exception {
-        assumeFalse("@AwaitsFix(bugUrl = https://github.com/elastic/elasticsearch/issues/45598)", Constants.WINDOWS);
         checkBlockedTrustManagerResource("truststore", "truststore.path");
     }
 
     public void testMessageForCertificateAuthoritiesOutsideConfigDir() throws Exception {
-        assumeFalse("@AwaitsFix(bugUrl = https://github.com/elastic/elasticsearch/issues/45598)", Constants.WINDOWS);
         checkBlockedTrustManagerResource("certificate_authorities", "certificate_authorities");
     }
 
@@ -231,7 +227,7 @@ public class SSLErrorMessageTests extends ESTestCase {
         exception = exception.getCause();
         assertThat(exception, throwableWithMessage(
             "failed to initialize SSL " + sslManagerType + " - access to read " + fileType + " file [" + fileName +
-                "] is blocked; SSL resources should be placed in the [" + env.configFile() + "] directory"));
+                "] is blocked; SSL resources should be placed in the [" + env.configFile().normalize() + "] directory"));
         assertThat(exception, instanceOf(ElasticsearchException.class));
 
         exception = exception.getCause();
@@ -253,7 +249,7 @@ public class SSLErrorMessageTests extends ESTestCase {
     }
 
     private String blockedFile() throws IOException {
-        return "/this/path/is/outside/the/config/directory/file.error";
+        return Paths.get("/this", "path", "is", "outside", "the", "config", "directory", "file.error").toString();
     }
 
     /**

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/ssl/SSLErrorMessageTests.java
@@ -9,6 +9,7 @@ import org.apache.lucene.util.Constants;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.ElasticsearchSecurityException;
 import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.io.PathUtils;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.env.TestEnvironment;
@@ -26,7 +27,6 @@ import java.nio.file.CopyOption;
 import java.nio.file.Files;
 import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.nio.file.StandardOpenOption;
 import java.nio.file.attribute.FileAttribute;
 import java.nio.file.attribute.PosixFilePermission;
@@ -249,7 +249,7 @@ public class SSLErrorMessageTests extends ESTestCase {
     }
 
     private String blockedFile() throws IOException {
-        return Paths.get("/this", "path", "is", "outside", "the", "config", "directory", "file.error").toString();
+        return PathUtils.get("/this", "path", "is", "outside", "the", "config", "directory", "file.error").toString();
     }
 
     /**

